### PR TITLE
add sync_log for kvrpc pb

### PR DESCRIPTION
**Pull request by javaforfun**

### CODE REVIEW

File: proto/raft_cmdpb.proto [hunk 1]
```diff
@@ -246,6 +246,8 @@ message RaftCmdRequest {
     repeated Request requests             = 2; 
     optional AdminRequest admin_request   = 3; 
     optional StatusRequest status_request = 4;
+
+    optional bool sync_log                = 5;
```
Comment (siddontang) [inline proto/raft_cmdpb.proto]: I prefer to move this to the request header.

Comment (javaforfun) [reply proto/raft_cmdpb.proto]: should we sync_log for admin_request?  eg. conf_change, split

Comment (siddontang) [reply proto/raft_cmdpb.proto]: I think we should do. 
Maybe we should also sync compact log, but I worry this may hurt the performance.

Comment (siddontang) [reply proto/raft_cmdpb.proto]: /cc @BusyJay

Comment (BusyJay) [reply proto/raft_cmdpb.proto]: I don't see any reason to sync compact log.

Comment (javaforfun) [reply proto/raft_cmdpb.proto]: we can distinguish admin_request type in tikv,  so it's no need to let pd known sync_log,  will move sync_log to request header.

**Comment by siddontang** [comment]
LGTM
Please fix the conflicts. 

PTAL @BusyJay

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**